File: Ruby (fail)/Terre/terre05.rb

 # Time :  15 min

 class Division
    attr_accessor   :dividend, :divider, :result, :rest
    def division

        @dividend   = ARGV[0].to_i
        @divider    = ARGV[1].to_i

        if (@divider == 0) || (@dividend < @divider) 
            puts "erreur."
        
        else
            @result = @dividend / @divider
            @rest   = @dividend - (@result * @divider)

            puts "résultat : #{@result}"
            puts "reste    : #{@rest}"
        end
    end
end

if __FILE__ == $0 
    dv = Division.new

    dv.division

end

File: Python/terre05.py
# Division
        
# Time = 10 min    
# Date = 29 aout 2023

import sys
import math


dividend   = int(sys.argv[1])
divider    = int(sys.argv[2])

print(dividend)
print(divider)

if divider == 0 or dividend < divider: 
    print ("erreur")
else :
    
    result = math.floor(dividend / divider)
    rest   = dividend - (result * divider)

    print ("résultat : ", result)
    print ("reste    : ", rest) 